src/snitch_section.cpp

```diff
@@ -15,12 +16,20 @@ section_entry_checker::~section_entry_checker() {
             // We are unwinding the stack because an exception has been thrown;
             // avoid touching the section state since we will want to report where
             // the exception was thrown.
+            state.reg.report_callback(
+                state.reg, event::section_ended{
+                               state.sections.current_section.back().id,
+                               state.sections.current_section.back().location, true});
```

Comment:
See comment in the approval tests; I think this is done too soon. When an exception is in flight, it has not been reported yet, but we now close the section immediately. This results in all unhandled exceptions being reported at the root of the test case, rather than inside the section from which it originated. 

I believe this particular case (close the section when unwinding an exception) can be handled in the registry instead, to solve this problem. That would be here: https://github.com/snitch-org/snitch/blob/d6632123cc8d13bdbc5cd60fd6741b9e0f635e82/src/snitch_registry.cpp#L554-L558

Reply:
Fixed.